src/Kiota.Builder/Writers/Php/PhpConventionService.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder.Writers.Php
+{
+    public class PhpConventionService: CommonLanguageConventionService
+    {
+        public override string TempDictionaryVarName => "urlTplParams";
+
+        public override string GetAccessModifier(AccessModifier access)
+        {
+            return (access) switch
+            {
+                (AccessModifier.Public) => "public",
+                (AccessModifier.Protected) => "protected",
+                _ => "private"
+            };
+        }
+        public override string StreamTypeName => "StreamInterface";
+
+        public override string VoidTypeName => "void";
+
+        public override string DocCommentPrefix => " * ";
+
+        private static string PathParametersPropertyName => "$pathParameters";
+
+        private static string RequestAdapterPropertyName => "$requestAdapter";
+
+        public override string ParseNodeInterfaceName => "ParseNode";
+
+        public string DocCommentStart => "/**";
+
+        public string DocCommentEnd => "*/";
+
+        public override string GetTypeString(CodeTypeBase code, CodeElement targetElement, bool includeCollectionInformation = true)
+        {
+            if(code is CodeUnionType) 
+                throw new InvalidOperationException($"PHP does not support union types, the union type {code.Name} should have been filtered out by the refiner.");
+            if (code.IsCollection || targetElement is CodeProperty propertyVar && propertyVar.IsOfKind(CodePropertyKind.PathParameters))
```

Comment:
why not switch the path parameter property to a collection with a refiner instead?